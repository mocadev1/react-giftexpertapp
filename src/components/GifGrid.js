import React, { useEffect, useState } from 'react'
import GifGridItem from './GifGridItem';

const GifGrid = ( { category } ) => {
    
    const [images, setImages] = useState( [] );
    
    useEffect( () => {
        getGifs();
    }, [] );
    
    const getGifs = async() => {

        const url = `https://api.giphy.com/v1/gifs/search?q=${ encodeURI(category) }&lang=es&limit=10&api_key=uwz8zqvnsCEQtNHMZLhPUxRpBzp877Xq`;
        const resp = await fetch( url );
        const { data } = await resp.json();

        const gifs = data.map( img => {
            return {
                id: img.id,
                title: img.title,
                url: img.images?.downsized_medium.url
            }
        });
        

        console.log(gifs);
        setImages(gifs);

        
    }
    
    
    // getGifs(); 
    
    
    return (
        <> 
            <h3> { category } </h3>
            <div className="card-grid">

                {
                    images.map( ( img ) => (
                        <GifGridItem
                            key={img.id}
                            { ...img }
                        />
                    ) )
                }

            </div>
        </>
    )
}

export default GifGrid
