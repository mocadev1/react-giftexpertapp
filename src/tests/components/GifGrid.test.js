import '@testing-library/jest-dom';

import { shallow } from 'enzyme';
import GifGrid from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Pruebas en el <GifGrid />', () => {
   
    const category = 'Nezuko chan';


    test('debe de renderizarse correctamente', () => {

        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });

        const wrapper = shallow( <GifGrid  category={ category } /> );

        expect(wrapper).toMatchSnapshot();
        
    });
    

    test('debe de mostrar items cuando se cargan imÃ¡genes useFetchGifs', () => {

        const gifs = [{
            id: 'ABC',
            url: 'https://localhost/some/thing.jp',
            title: 'Un titulo de test'
        },
        {
            id: 'ABC',
            url: 'https://localhost/some/thing.jp',
            title: 'Un titulo de test'
        }];
        
        
        
        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });
        
        const wrapper = shallow( <GifGrid category={ category} /> );

        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('p').exists() ).toBe(false);
        expect( wrapper.find('GifGridItem').length ).toBe( gifs.length );

        console.log(gifs.length);
        
    });

    // test('debe de contener a lo menos un <GifGridItem />', () => {
        
    // });
    
    
    
    
    
    
});
