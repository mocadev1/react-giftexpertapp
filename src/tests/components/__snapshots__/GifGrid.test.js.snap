// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el <GifGrid /> debe de mostrar items cuando se cargan im√°genes useFetchGifs 1`] = `
<Fragment>
  <h3>
     
    Nezuko chan
     
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Un titulo de test"
      url="https://localhost/some/thing.jp"
    />
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Un titulo de test"
      url="https://localhost/some/thing.jp"
    />
  </div>
</Fragment>
`;

exports[`Pruebas en el <GifGrid /> debe de renderizarse correctamente 1`] = `
<Fragment>
  <h3>
     
    Nezuko chan
     
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
